src/ORM/Table.php

```diff
@@ -1011,6 +1011,14 @@ public function findAll(Query $query, array $options)
      */
     public function findList(Query $query, array $options)
     {
+        if (!isset($options['keyField']) && !isset($options['valueField'])) {
+            $select = $query->clause('select');
+            if ($select && count($select) <= 2) {
+                $options['keyField'] = array_shift($select);
+                $options['valueField'] = array_shift($select) ?: $options['keyField'];
```

Comment:
the query has not been executed at this point, but that is actually not the problem here. What I meant is that the fields syntax as accepted by select() is different from what keyField and valueField accepts.

If you want to try it yourself, add a test where you use 'Articles.id' as one of the fields in select()